### ISSUE TITLE
Mitigate crash risk and provide a way to propagate lightstep exceptions to consumer

### ISSUE THREAD

**Issue by kirkplangrid** [lightstep/lightstep-tracer-csharp#68]
We've had several crashes of our Windows app due to exceptions being thrown in the tracer.  Forwarding these errors on to lightstep support provides the ability to fix these underlying issues.  However, it would be better if we could collaborate on improving the tracer without impacting our users (app crashes).

The proposal here then is that lightstep should ironcladly protect the app from exceptions in its own code by ensuring all their code is surrounded by a `try/catch`.  However, to ensure lightstep is still kept in the loop on underlying issues (so that they can get fixed) a callback should be provided to the tracer when its being set up.  Something like:

```
    .WithExceptionHandler(exception => exception.ReportToBugSnag());
```

The parameter would be of type `Action<Exception>` and perhaps any other accompanying data lightstep would find useful that we could send on its way to whatever crash reporter the consumers are using.

**Comment by austinlparker** [comment]
Thanks for the suggestion! We've released v0.4.3 that catches exceptions when translating spans to reports, but I like the exception handler idea a lot and will see how feasible that'll be to get in for our next minor.

**Comment by austinlparker** [comment]
Hey @kirkplangrid, can you take a look at that PR and see if it'll satisfy this?
